docs/usage.md

```diff
@@ -0,0 +1,309 @@
+- [Publishing using Apache Maven](#Publishing-using-Apache-Maven)
+- [Publishing using Gradle](#Publishing-using-Gradle)
+
+See [action.yml](../action.yml) for more details on task inputs.
+
+## Selecting Java distribution
+Input `distribution` is mandatory and should be provided to use action. See [Supported distributions](../README.md#Supported-distributions) for the list of available options.
+
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: '<distribution>' # See 'Supported distributions' for available distribution
+    java-version: '11.x'
+- run: java -cp java HelloWorldApp
+```
+
+### Supported version syntax
+Input `java-version` supports version range or exact version in [SemVer](https://semver.org/) format:
+- major versions: `8`, `11`, `15`, `11.x`
+- more specific versions: `8.0.232`, `11.0.4`, `11.0`, `11.0.x`
+- an early access (EA) versions: `15-ea`, `15.0.0-ea`, `15.0.0-ea.2`
+- legacy 1.x syntax: `1.8` (same as `8`), `1.8.0.212` (same as `8.0.212`)
+
+TO-DO: Do we really want to support this `1.` syntax? Is it a time to drop old syntax in V2?  
+TO-DO: Clarify docs about using syntax with 4 digits  
+
+### Zulu
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: 'zulu'
+    java-version: '11.x'
+    java-package: jdk # optional (jdk or jre) - defaults to jdk
+- run: java -cp java HelloWorldApp
+```
+
+### Adoptium
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: 'adoptium'
+    java-version: '11.x'
+- run: java -cp java HelloWorldApp
+```
+
+## Installing custom Java package type
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: '<distribution>'
+    java-version: '11.x'
+    java-package: jdk # optional (jdk or jre) - defaults to jdk
+- run: java -cp java HelloWorldApp
+```
+
+
+## Installing custom Java architecture
+
+```yaml
+steps:
+- uses: actions/checkout@v2
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: '<distribution>'
+    java-version: '11.x'
+    architecture: x86 # optional - defaults to x64
+- run: java -cp java HelloWorldApp
+```
+
+## Installing Java from local file
+If your use-case requires using custom distribution or version that is not provided in action by default, you can download it manually and then action will take care about installation and caching version on VM:
+
+TO-DO: Fix example
+```yaml
+steps:
+- run: |
+    download_url="https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.10%2B9/OpenJDK11U-jdk_x64_mac_hotspot_11.0.10_9.pkg"
+    wget $download_url
+- uses: actions/setup-java@v2-preview
+  with:
+    distribution: 'jdkFile'
+    jdkFile: runner.temp # Optional - jdkFile to install java from. Useful for distributions not provided by default
```

Comment:
```suggestion
    jdkFile: runner.temp # Optional - jdkFile to install java from.
```
removing it because we already said above about distributions not provided by action